### PR TITLE
Project language update fails on null #8014

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyLanguageParamsJson.java [hunk 1]
```diff
@@ -17,7 +18,10 @@
     ModifyLanguageParamsJson( @JsonProperty("name") final String name, @JsonProperty("language") final String language )
     {
         this.name = ProjectName.from( name );
-        this.language = Locale.forLanguageTag( language );
+        this.language = Optional.ofNullable( language ).
+            filter( lang -> !lang.trim().isEmpty() ).
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyLanguageParamsJson.java:22]: !isBlank()

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ProjectResource.java [hunk 2]
```diff
@@ -97,8 +97,9 @@ public ProjectJson modify( final MultipartForm form )
     @Path("modifyLanguage")
     public String modifyLanguage( final ModifyLanguageParamsJson params )
     {
-        return doApplyLanguage( params.getName(), params.getLanguage() ).
-            toLanguageTag();
+        return Optional.ofNullable( doApplyLanguage( params.getName(), params.getLanguage() ) ).
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ProjectResource.java:100]: To improve readability return optional from doApplyLanguage

File: modules/core/core-api/src/main/java/com/enonic/xp/content/Content.java [hunk 3]
```diff
@@ -444,31 +444,31 @@ protected Builder()
...
-            this.thumbnail = source.thumbnail;
-            this.childOrder = source.childOrder;
-            this.permissions = source.permissions;
-            this.inheritPermissions = source.inheritPermissions;
-            this.language = source.language;
-            this.contentState = source.contentState;
-            this.publishInfo = source.publishInfo;
-            this.processedReferences = ContentIds.create().addAll( source.processedReferences );
-            this.workflowInfo = source.workflowInfo;
+            this.id = source.getId();
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/Content.java:447]: getters to class of itself look very suspicions. What was wrong with direct field access?

File: modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java [hunk 4]
```diff
@@ -105,7 +103,7 @@ private Builder( final ProjectPermissions source )
 
         public Builder addOwner( final String owner )
         {
-            if ( StringUtils.isNotBlank( owner ) )
+            if ( owner != null )
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java:106]: this is not exact conversion. Exact one is `!nullToEmpty( owner ).isBlank()`
But why do we need this null guard? What cand of caller tries to add `null` owner/editor/author/...?

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5339458&bid=17478696)

**Pull request merged**